tests/core/pyspec/eth2spec/test/phase1/sanity/test_shard_blocks.py

```diff
@@ -0,0 +1,165 @@
+from eth2spec.test.context import (
+    PHASE0,
+    always_bls,
+    expect_assertion_error,
+    spec_state_test,
+    with_all_phases_except,
+)
+from eth2spec.test.helpers.shard_block import (
+    build_shard_block,
+    sign_shard_block,
+)
+from eth2spec.test.helpers.shard_transitions import is_full_crosslink
+from eth2spec.test.helpers.state import transition_to_valid_shard_slot
+
+
+def run_shard_blocks(spec, shard_state, signed_shard_block,
+                     beacon_parent_state,
+                     validate=True, valid=True):
+    pre_shard_state = shard_state.copy()
+    yield 'pre', pre_shard_state
+    yield 'signed_shard_block', signed_shard_block
+    yield 'validate', validate
+    yield 'beacon_parent_state', beacon_parent_state
+
+    if validate is False:
+        beacon_parent_state = None
+
+    if not valid:
+        expect_assertion_error(lambda: spec.shard_state_transition(
+            shard_state, signed_shard_block, validate=validate, beacon_parent_state=beacon_parent_state)
+        )
+        yield 'post', None
+        return
+
+    spec.shard_state_transition(shard_state, signed_shard_block,
```

Comment:
I'm still unsure of the optional `beacon_parent_state` and the optional `validate`  in `shard_state_transition`.  IMO, validations should occur in the state transition (even (and maybe especially) when creating the block). The only validations that should be able to be skipped should be the ones that rely upon "results" or values from the  post-processing (e.g. block signature and state root)

Everything in `verify_shard_block_message` can and should be validated no matter what, and thus `beacon_parent_state` should be required.

We can add a `validate_results` that can enabled/disable sig verification and shard state root check (we don't embed shard state root in the block right now but probably should)

Reply:
If we want to make `beacon_parent_state` required, we can call `process_shard_block` instead of `shard_state_transition` in [get_shard_transition_fields](https://github.com/ethereum/eth2.0-specs/blob/661959c13ec422cbfb588d40f0cb57fd9ff5c5fa/specs/phase1/validator.md#shard-transition).

((Updating it now))